ls1intum/artemis-android
# `Thesis`: Add Survey dialog

### Problem Description
As part of my masters thesis, I want to conduct a survey of the Artemis Android app users. To make the users aware of that survey, I will display a small hint in the course overview for the duration of the survey (approx. 2 weeks). The survey will be hosted by the chair owned domain https://survey.ase.in.tum.de/. 

Here are the preliminary questions: https://confluence.ase.in.tum.de/spaces/ArTEMiS/pages/232761693/Artemis+Android+Questionnaire

### Changes
 
- Added survey hint in the CourseOverview
- Added survey dialog
- Restructured CourseOverview code
  - extracted BetaHintDialog
  - extracted CourseList


### Steps for testing
- Go to the course overview page
- Notice the survey hint
- Click on it
- Press "Not now" -> dialog closes
- Click on it again 
- Press "Participate" -> for now this only takes you too a example domain in the browser
- Navigate back to the app -> the hint is not shown anymore


### Screenshots
![image](https://github.com/user-attachments/assets/dcdd998e-a0d5-44a5-94af-fbd63c5d5d1e)
![image](https://github.com/user-attachments/assets/03904d3a-f0ee-4b84-87ff-5ae01d0b4b26)


## Next steps
As soon as the survey questions and timeline is finalized
- Add appropriate start and end dates (the survey hint will only be displayed during that period) -> DONE
- Replace the sample with the real survey link -> DONE
- Merge the PR
- Create a new release
